src/crypto/tls/generate_cert.go

```diff
@@ -136,6 +137,11 @@ func main() {
 		template.KeyUsage |= x509.KeyUsageCertSign
 	}
 
+	if *isDC { //TODO: should this check that it is not a CA?
```

Comment:
No check should be necessary, someone could possibly create a self-signed certificate and somehow perform key pinning to ensure that this particular certificate is trusted.

The CA basic constraint is documented here: https://tools.ietf.org/html/rfc5280#section-4.2.1.9
It only affects checking other certificate signatures, it should not affect validation of the TLS handshake signature (which is the only signature covered by a DC).

Reply:
Thank you for the info!